src/main/java/reposense/report/ContributionSummaryGenerator.java

```diff
@@ -14,40 +14,30 @@
 import reposense.dataobject.CommitInfo;
 import reposense.dataobject.RepoConfiguration;
 import reposense.dataobject.RepoContributionSummary;
-import reposense.dataobject.RepoInfo;
 
 
 public class ContributionSummaryGenerator {
 
-    public static Map<String, RepoContributionSummary> analyzeContribution(
-            List<RepoInfo> repos, List<RepoConfiguration> configs) {
-        System.out.println("Generating summary report...");
-        Map<String, RepoContributionSummary> result = new HashMap<>();
-        HashSet<Author> suspiciousAuthors = new HashSet<>(); //authors with bugs that I didnt catch
-        Date startDate = configs.get(0).getFromDate() == null ? getStartDate(repos) : configs.get(0).getFromDate();
-        for (RepoInfo repo : repos) {
-            //if (repo.getCommits().isEmpty()) continue;
-            RepoContributionSummary summary = new RepoContributionSummary(repo);
-            summary.setFromDate(startDate);
-            summary.setToDate(configs.get(0).getToDate());
-            summary.setAuthorWeeklyIntervalContributions(
-                    getAuthorIntervalContributions(repo, startDate, 7, suspiciousAuthors));
-            summary.setAuthorDailyIntervalContributions(
-                    getAuthorIntervalContributions(repo, startDate, 1, suspiciousAuthors));
-            summary.setAuthorFinalContributionMap(repo.getAuthorContributionMap());
-            summary.setAuthorContributionVariance(
-                    calcAuthorContributionVariance(summary.getAuthorDailyIntervalContributions()));
-            summary.setAuthorDisplayNameMap(repo.getAuthorDisplayNameMap());
-            result.put(repo.getDirectoryName(), summary);
-        }
-        if (!suspiciousAuthors.isEmpty()) {
-            System.out.println("PLEASE NOTE, BELOW IS THE LIST OF SUSPICIOUS AUTHORS:");
-            for (Author author : suspiciousAuthors) {
-                System.out.println(author);
-            }
-        }
-        System.out.println("done!Congrats!");
-        return result;
+    /**
+     * Analyzes contribution of authors in a single repository.
+     * @param suspiciousAuthors authors with bugs not caught in the analyzer.
+     */
+    public static RepoContributionSummary analyzeContribution(
+            RepoConfiguration config,
+            List<CommitInfo> commitInfos,
+            HashMap<Author, Integer> authorContributionMap,
+            HashSet<Author> suspiciousAuthors) {
+        Date startDate = config.getFromDate() == null ? getStartDate(commitInfos) : config.getFromDate();
+        RepoContributionSummary summary = new RepoContributionSummary();
+        summary.setAuthorWeeklyIntervalContributions(
+                getAuthorIntervalContributions(config, commitInfos, startDate, 7, suspiciousAuthors));
```

Comment:
previously mentioned to remove the weekly and do it in the frontend instead? in this PR or open new?

Reply:
In another PR instead